### PR TITLE
Feat(install tabs): introduce

### PR DESCRIPTION
**Pull request by AugustinMauroy**
## Description

Adding `installTabs`

In nodejs.dev the style of installTabs is not the same because style is "sparated" on par in `installTabs` others part is on `pages/index.tsx` so I have merge the style to have this component better.
But I don't have take leafs in this component

## Related Issues

Related issues: [Migrate DownloadComponents and Hero Component from nodejs.dev #5303](https://github.com/nodejs/nodejs.org/issues/5303)

### Check List

- [X] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [X] I have run `npx turbo lint` to ensure the code follows the style guide. And run `npx turbo lint:fix` to fix the style errors if necessary.
- [X] I have run `npx turbo format` to ensure the code follows the style guide.
- [X] I have run `npx turbo test` to check if all tests are passing, and/or `npx turbo test:snapshot` to update snapshots if I created and/or updated React Components.
- [ ] I've covered new added functionality with unit tests if necessary.

### CODE REVIEW

File: components/Home/InstallTabs/LinuxPanel/index.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      </LocalizedLink>
+    </div>
+  );
+};
+
+const LinuxPanel: FC = () => {
+  const [nvmVersion, setNvmVersion] = useState('');
+
+  useEffect(() => {
+    getNvmLatestVersion().then(version => setNvmVersion(version));
```
Comment (Harkunwar) [inline components/Home/InstallTabs/LinuxPanel/index.tsx:43]: I would recommend if we use the TanStack Query library for handling all async code, it does error handling really well and prevents us from pitfalls of `useEffect`.

Comment (AugustinMauroy) [reply components/Home/InstallTabs/LinuxPanel/index.tsx:43]: This lib in't on codebase. So I will take a look to SWR

File: components/Home/InstallTabs/index.tsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+  MAC: [os.mac, os.win, os.linux],
+  LINUX: [os.linux, os.mac, os.win],
+  OTHER: [os.win, os.mac, os.linux],
+};
+
+const InstallTabs: FC = () => {
+  const [reactTabs, setReactTabs] = useState<ReactNode>();
+
+  useEffect(() => {
+    const userOS = detectOS();
```
Comment (Harkunwar) [inline components/Home/InstallTabs/index.tsx:61]: Could you use the hook `useDetectOS` instead?

Comment (AugustinMauroy) [reply components/Home/InstallTabs/index.tsx:61]: Yup true

Comment (Harkunwar) [reply components/Home/InstallTabs/index.tsx:61]: @AugustinMauroy Actually if you don't need the download link i think directly using it is okay as it doesn't really hold state. So I think what you did is fine!

Comment (AugustinMauroy) [reply components/Home/InstallTabs/index.tsx:61]: And I have found an thinks we have to investigate.
I'm on macOS + safari and he output windows ...

File: components/Home/InstallTabs/LinuxPanel/index.tsx (hunk 1 above)
Comment (AugustinMauroy) [reply components/Home/InstallTabs/LinuxPanel/index.tsx:43]: what do you thinks about this code:
```js
 const [nvmVersion, setNvmVersion] = useState<string | null>(null);
  const { data: nvmLatestVersion } = useSWR('nvm-latest-version', () =>
    getNvmLatestVersion()
  );

  useEffect(() => {
    if (nvmLatestVersion) {
      setNvmVersion(nvmLatestVersion);
    }
  ```
  personally I don't found that better

**Comment by ovflowd** [comment]
@AugustinMauroy the builds are failing, just as a fyi

File: components/Home/InstallTabs/index.tsx [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+import type { UserOS } from '../../../types';
+import type { FC, ReactNode } from 'react';
+
+const getOSPanel: FC<UserOS> = userOS => {
+  switch (userOS) {
+    case 'MAC':
+      return (
+        <>
+          <TabPanel
+            className={styles.reactTabsTabPanel}
```
Comment (mikeesto) [inline components/Home/InstallTabs/index.tsx]: Small suggestion - perhaps create a styles object:

```tsx
const tabPanelStyles = {
  className: styles.reactTabsTabPanel,
  selectedClassName: styles.reactTabsTabPanelSelected
};
  ```
  
  And spread it into each tab panel: `<TabPanel {...tabPanelStyles}>`

Comment (AugustinMauroy) [reply components/Home/InstallTabs/index.tsx]: Done thanks for the idea!

**Comment by ovflowd** [comment]
**FYI:** We deleted the `major/website-redesign` branch to simplify our release and rollout process. Please feel free to recreate this PR or reopen it with the `main` branch as a target.

Thank you, and we apologise for the inconvenience!

**Pull request closed without merging**